libs/agno/agno/agent/agent.py

```diff
@@ -5390,6 +5548,114 @@ def _add_reasoning_metrics_to_extra_data(self, reasoning_time_taken: float) -> N
 
             log_error(f"Failed to add reasoning metrics to extra_data: {str(e)}")
 
+    def _get_effective_filters(self, knowledge_filters: Optional[Dict[str, Any]] = None) -> Optional[Dict[str, Any]]:
+        """
+        Determine which knowledge filters to use, with priority to run-level filters.
+
+        Args:
+            knowledge_filters: Filters passed at run time
+
+        Returns:
+            The effective filters to use, with run-level filters taking priority
+        """
+        effective_filters = None
+
+        # If agent has filters, use those as a base
+        if self.knowledge_filters:
+            effective_filters = self.knowledge_filters.copy()
+
+        # If run has filters, they override agent filters
+        if knowledge_filters:
+            if effective_filters:
+                # Merge filters, with run filters taking priority
+                effective_filters.update(knowledge_filters)
+            else:
+                effective_filters = knowledge_filters
+
+        if effective_filters:
+            log_debug(f"Using knowledge filters: {effective_filters}")
+
+        return effective_filters
+
+    def _get_potential_filters_response(self, message: str) -> ModelResponse:
```

Comment:
Hmm why don't we just give all the potential filters to the agent in the system message, and then the agent specifies the filter to use when it calls the search_knowledge function? This feels a bit over-engineered.